dbms/src/Storages/DeltaMerge/Index/InvertedIndex/Reader.cpp

```diff
@@ -152,9 +154,62 @@ void InvertedIndexMemoryReader<T>::load(ReadBuffer & read_buf, size_t index_size
     }
 }
 
+namespace
+{
+
+template <typename T>
+inline bool isKeyOutOfRange(const InvertedIndexReader::Key key)
+{
+    if constexpr (std::is_signed_v<T>)
+    {
+        using SignedKey = std::make_signed_t<InvertedIndexReader::Key>;
+        SignedKey signed_key = key;
+        return signed_key < std::numeric_limits<T>::min() || signed_key > std::numeric_limits<T>::max();
+    }
+    else
+    {
+        return key > std::numeric_limits<T>::max();
+    }
+}
+
+template <typename T>
+inline bool isKeyLessThanMin(const InvertedIndexReader::Key key)
+{
+    if constexpr (std::is_signed_v<T>)
+    {
+        using SignedKey = std::make_signed_t<InvertedIndexReader::Key>;
+        SignedKey signed_key = key;
+        return signed_key < std::numeric_limits<T>::min();
+    }
+    else
+    {
+        return false;
+    }
+}
+
+template <typename T>
+inline bool isKeyGreaterThanMax(const InvertedIndexReader::Key key)
+{
+    if constexpr (std::is_signed_v<T>)
+    {
+        using SignedKey = std::make_signed_t<InvertedIndexReader::Key>;
+        SignedKey signed_key = key;
+        return signed_key > std::numeric_limits<T>::max();
+    }
+    else
+    {
+        return key > std::numeric_limits<T>::max();
```

Comment:
There types are not floating types, so in what cases would `key > std::numeric_limits<T>::max()` happen?

Reply:
`key` is always `UInt64`.

Reply:
It is used to fix this case: https://github.com/pingcap/tiflash/pull/10048/commits/e56441dc2574f55e76bab9729871901398f212ee